File: fixtures/polyglot/java/com/acme/orders/PaymentService.java
package com.acme.orders;

import com.acme.billing.InvoiceRepository;
import com.acme.identity.UserContext;

public class PaymentService implements PaymentPort {
  private InvoiceRepository invoices;
  private UserContext context;

  public PaymentReceipt charge(PaymentRequest request) {
    return invoices.create(request);
  }
}

interface PaymentPort {
  PaymentReceipt charge(PaymentRequest request);
}

File: fixtures/polyglot/swift/PaymentService.swift
import Foundation

protocol PaymentPort {
  func charge(request: PaymentRequest) -> PaymentReceipt
}

struct PaymentRequest {
  let orderId: String
}

struct PaymentReceipt {
  let id: String
}

final class PaymentService: PaymentPort {
  private let invoices: InvoiceRepository

  init(invoices: InvoiceRepository) {
    self.invoices = invoices
  }

  func charge(request: PaymentRequest) -> PaymentReceipt {
    return invoices.create(request: request)
  }
}

final class InvoiceRepository {
  func create(request: PaymentRequest) -> PaymentReceipt {
    return PaymentReceipt(id: request.orderId)
  }
}
